cogs/Announce.py

```diff
@@ -0,0 +1,418 @@
+        return temp
+
+    def get_receivers(self, guild_id, roles):
+        """
+        A function to get the receivers of a particular announcement
+        :param roles: The list of roles in the guild
+        :param guild_id: The id of the guild
+        :return: All the receivers of the announcement
+        """
+        temp = []
+        for role in self.roles[guild_id]:
+            temp += discord.utils.get(roles, id=role).members
+        return list(set(temp))
+
+    def receiver_msg(self, guild):
+        """
+        A function to create a string message about receivers
+        :param guild: The guild of the bot
+        :return: A string message about receivers
+        """
+        if not self.roles[guild.id]:
+            return f"You are currently sending to Everyone and there are {str(len(guild.members))} receivers"
+        return f"You are currently sending to {self.get_role_names(guild.id, guild.roles)} and there are {str(len(self.get_receivers(guild.id, guild.roles)))} receivers "
+
+    def construct_embed(self, guild: discord.Guild):
+        """
+        Constructing an embedded message from the information stored in the manager
+        :param guild: The the guild
+        :return: An embedded message for the announcement
+        """
+        message = self.messages[guild.id]
+        embed: discord.Embed = discord.Embed(title=message.title,
+                                             description=message.description, colour=KoalaColours.KOALA_GREEN)
+        embed.set_author(name="announcement from " + guild.name)
+        if message.thumbnail != 'https://cdn.discordapp.com/':
+            embed.set_thumbnail(url=message.thumbnail)
+        return embed
+
+    @commands.group(name="announce")
+    async def announce(self, ctx):
+        """
+        Use k!announce create to create an announcement
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.send(f"Please use `{KoalaBot.COMMAND_PREFIX}help announce` for more information")
+
+    @commands.check(announce_is_enabled)
+    @announce.command(name="create")
+    async def create(self, ctx):
+        """
+        This command creates a new message that will be available for sending
+        :param ctx: The context of the bot
+        :return:
+        """
+        if not self.not_exceeded_limit(ctx.guild.id):
+            remaining_days = math.ceil(ANNOUNCE_SEPARATION_DAYS-((int(time.time()) - self.announce_database_manager.get_last_use_date(
+                ctx.guild.id)) / SECONDS_IN_A_DAY))
+            await ctx.send("You have recently sent an announcement and cannot use this function for " + str(remaining_days) + " days")
+            return
+        if self.has_active_msg(ctx.guild.id):
+            await ctx.send("There is currently an active announcement")
+        else:
+            await ctx.send("Please enter a message")
+            message, channel = await wait_for_message(self.bot, ctx)
```

Comment:
**Everywhere** where you have a `wait_for_message` you need to tell the user you are waiting for x amount of time
```suggestion
            await ctx.send("Please enter a message. I'll wait 60 seconds, don't worry.")
            message, channel = await wait_for_message(self.bot, ctx)
```